### PR TITLE
[W5.7][F10-4] ANG ZHI KAI

### PR DESCRIPTION
**Pull request by azhikai**
Added `sort` command.

### CODE REVIEW

**Comment by azhikai** [comment]
**Still WIP!**

Commit 406f1dd failed due to an erroneous change in commit 052d710 where I accidentally included the `sort` into the `HelpCommand.java`. Naturally it will cause the CI build to fail.

More work can be added into this PR by adding test cases for the SortCommand, but I'll leave that out for now until I have more time to look into it.

**Comment by azhikai** [comment]
Added unit test for the `sort` command. Along with that, `input.txt` and `expected.txt` has been updated as well.

Did a minor modification to `runtests.bat` so the terminal don't close immediately after doing `FC`.

No more commits will be made for this LO. **Ready for review!**

**Comment by rebstan97** [comment]
Great work! But can perhaps update ParserTest, adding a test to check if the sort command is parsed correctly.

**Comment by yican95** [comment]
Good Job. Can add a test to test sort command on empty addressBook

**Comment by azhikai** [comment]
@rebstan97 @yican95 
Thanks both for your review. Will work on that. 👍

**Comment by m4dkip** [comment]
Well done, all relevant classes and tests updated.

**Comment by HyperionNKJ** [comment]
Great job! Really like how you updated everything, even the google docs. May consider extending sort to tags.

**Comment by wn** [comment]
Commit message shouldn't have full stops.

**Comment by azhikai** [comment]
> Commit message shouldn't have full stops.

Good catch. Updated to remove the full stop from the title.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -156,6 +156,11 @@ Views all details of the 1st person in the results of the `find` command.
 Clears all entries from the address book. +
 Format: `clear`
 
+== Sorting the entries : `sort`
+
+Sort the entries by name in alphabetical order (ASC) +
```
Comment (dalessr) [inline docs/UserGuide.adoc:161]: Minor nit but the description should be aligned with others (i.e. `Sorts the entries...`, full stop)

File: src/seedu/addressbook/commands/HelpCommand.java [hunk 2]
```diff
@@ -8,21 +8,23 @@
 
     public static final String COMMAND_WORD = "help";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows program usage instructions.\n"
-            + "Example: " + COMMAND_WORD;
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Shows program usage instructions.\n"
+                    + "Example: " + COMMAND_WORD;
```
Comment (dalessr) [inline src/seedu/addressbook/commands/HelpCommand.java:13]: This doesn't seem to follow the coding standards. You can move the second line to the first line as long as it's not reaching the limit.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -4,14 +4,13 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.Iterator;
 import java.util.List;
-
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:9]: This line shouldn't be removed as we are importing packages from different sources (one external one internal). It doesn't follow the coding standards if you combine them into one group.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 4]
```diff
@@ -24,23 +23,27 @@
...
+
         protected DuplicatePersonException() {
             super("Operation would result in duplicate persons");
         }
     }
 
     /**
-     * Signals that an operation targeting a specified person in the list would fail because
-     * there is no such matching person in the list.
+     * Signals that an operation targeting a specified person in the list would fail because there
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:33]: Hmm is this change really necessary? Try to make the least changes to achieve your goal every time in order to make your PR clean & clear.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -24,23 +23,27 @@
...
+     * is no such matching person in the list.
      */
-    public static class PersonNotFoundException extends Exception {}
+    public static class PersonNotFoundException extends Exception {
+
+    }
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:38]: The original code is correct as there's no implementation inside the method.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 6]
```diff
@@ -24,23 +23,27 @@
...
 
     private final List<Person> internalList = new ArrayList<>();
 
     /**
      * Constructs empty person list.
      */
-    public UniquePersonList() {}
+    public UniquePersonList() {
+    }
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:46]: Same as above

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 7]
```diff
@@ -99,9 +103,9 @@ public boolean contains(ReadOnlyPerson toCheck) {
     /**
      * Adds a person to the list.
      *
-     * @throws DuplicatePersonException if the person to add is a duplicate of an existing person in the list.
-     *    The @link{ReadOnlyPerson#isSamePerson} method is used for this comparison,
-     *    which defines a weaker notion of equality.
+     * @throws DuplicatePersonException if the person to add is a duplicate of an existing person in
+     * the list. The @link{ReadOnlyPerson#isSamePerson} method is used for this comparison, which
+     * defines a weaker notion of equality.
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:108]: Hmm I think there should be spaces for the second line & third line of `@throws` (aligned with `If the person`)

https://www.oracle.com/technetwork/java/javase/tech/index-137868.html

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 8]
```diff
@@ -138,6 +151,6 @@ public void clear() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof UniquePersonList // instanceof handles nulls
-                        && this.internalList.equals(((UniquePersonList) other).internalList));
+                && this.internalList.equals(((UniquePersonList) other).internalList));
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:154]: This change violates the coding standard (are `||` and `&&` at the same level here?)

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
+package seedu.addressbook.commands;
```
Comment (dalessr) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:1]: Good job adding JUnit test for the new feature 👍 . It's better to add/modify tests for other relevant classes also (i.e. `parser`)

**Review by dalessr — APPROVED** [review]
@AZhiKai Well done for the `sort` command 👍 

Please close the PR after looking at the comments below.

File: src/seedu/addressbook/commands/HelpCommand.java (hunk 2 above)
Comment (azhikai) [reply src/seedu/addressbook/commands/HelpCommand.java:13]: I think it got auto wrapped due to the limit I set which is at 100 chars. Will look into it again. Thanks!

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 4 above)
Comment (azhikai) [reply src/seedu/addressbook/data/person/UniquePersonList.java:33]: I think it accidentally auto formatted the whole file when scrolling through. Same reasons apply to the next few.

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 8 above)
Comment (azhikai) [reply src/seedu/addressbook/data/person/UniquePersonList.java:154]: Caused by auto formatting as well. Will look into my Java code style settings in awhile.
